src/garbage.rs

```diff
@@ -39,9 +39,6 @@ pub struct Garbage {
     func: Deferred,
 }
 
-unsafe impl Sync for Garbage {}
-unsafe impl Send for Garbage {}
```

Comment:
I would keep this impl as it encodes the assertion that we need `Garbage` to be sendable across threads.